Sort leaderboard by name

chigby opened the pull request:
Fixes #278 

This pull request changes the sorting of the leaderboard to sort by news organization name after sorting by the score.

I've chosen to add `lodash` as a dependency here since its sorting capabilities are more flexible and more in line with the kind of sorting we want to perform here. This doesn't even really add that much to the "weight" of our JS. According to [the documentation](https://lodash.com/per-method-packages), if we just import the single function we want, it will only even add the small subset of lodash code that we're using into the final bundle.

chigby committed:
Add `lodash` as a dependency

We're right now sorting the leaderboard using `underscore`, which has
more limited sorting capabilities than `lodash`.

chigby committed:
leaderboard: Add secondary sort by name

SaptakS reviewed (approved):
Looks good to me. I am not 100% sure if removing the article is a good idea, but I feel we will be able to know that only after more user testing whether that confuses people or if that's what they expect.

chigby commented:
Yeah, I suppose not considering articles when sorting by name is maybe a question for @redshiftzero. Initially I didn't think about it, but when I was scrolling through locally, I noticed all the organizations with "The" at the beginning were next to each other. I think it depends on how we think people want to use the list and if, for example, "The New York Times" should be under "T" or "N".

redshiftzero commented:
nice, I think removing articles is a good choice since we have so many news organizations beginning with The (and this seems to be a [standard approach](https://en.wikipedia.org/wiki/Alphabetical_order#The_and_other_common_words))

SaptakS commented:
@chigby I have resolved the merge conflict modifying my code for sorting by different attributes based on your changes in the PR. Please review and check if looks good. If seems good to you, can merge then

chigby committed:
Fix alphabetical sorting bug

Oops, the previous regular expression here was actually replacing the
letter "A" at the beginning of any site name when it was not followed
by a space, meaning "ABC News" was being sorted as "BC News."

chigby commented:
@SaptakS Your change looks great, though while I was looking at it locally I realized there was a problem with my title sorting regular expression. I fixed it in https://github.com/freedomofpress/securethenews/pull/290/commits/ae45db103523ff579ec6ed384a0518ee5284a375 but if you could give it a quick check to make sure, then we can get this merged.

SaptakS commented:
Regex fix looks good. Nice catch!
